app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -40,7 +40,11 @@ def check_user_enabled
   end
 
   def authorized
-    User.current.allowed_to?(path_to_authenticate)
+    if @available_sso.respond_to?(:allowed_to?) && @available_sso&.scope&.any?
+      @available_sso.allowed_to?(path_to_authenticate) && User.current.allowed_to?(path_to_authenticate)
+    else
+      User.current.allowed_to?(path_to_authenticate)
+    end
```

Comment:
Are there any other SSO methods other than JWT? This could break them.

Reply:
I believe this should go into the SSO classes. Let's keep this concern small. I could imagine an API like this:

```ruby
if @available_sso.supports?(:scopes)
if @available_sso.has_feature?(:scopes)
```